src/core-services/orders/queries/orders.js

```diff
@@ -42,10 +41,8 @@ export default async function orders(context, { filters, shopIds } = {}) {
   // If an admin wants all orders for an account, we force it to be limited to the
   // shops for which they're allowed to see orders.
   if (shopIds) {
-    shopIds.forEach((shopId) => {
-      if (!userHasPermission(["orders", "order/fulfillment"], shopId)) {
-        throw new ReactionError("access-denied", "Access Denied");
-      }
+    shopIds.forEach(async (shopId) => {
+      await checkPermissions(["orders", "order/fulfillment"], shopId);
```

Comment:
Is this OK to do in a `forEach`, or does it need to be done in a `for...of` instead?